server/src/main/java/org/eclipse/lsp/cobol/core/model/variables/AbstractVariable.java

```diff
@@ -16,23 +16,40 @@
 package org.eclipse.lsp.cobol.core.model.variables;
 
 import org.eclipse.lsp.cobol.core.model.Locality;
-import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NonNull;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 
 /** This abstract class implements the functionality that is common for all the variables. */
-@AllArgsConstructor
 public abstract class AbstractVariable implements Variable {
   @Getter protected final int levelNumber;
   @Getter protected final String name;
   @Getter protected final Locality definition;
+  @Getter protected final boolean global;
   private final List<Locality> usages = new ArrayList<>();
   private final List<ConditionDataName> conditionChildren = new ArrayList<>();
-  private Variable parent;
+  private final Variable parent;
+
+  protected AbstractVariable(int levelNumber, String name, Locality definition, Boolean global, Variable parent) {
+    this.levelNumber = levelNumber;
+    this.name = name;
+    this.definition = definition;
+    this.global = Optional.ofNullable(parent).map(Variable::isGlobal)
+        .orElseGet(() -> Optional.ofNullable(global).orElse(false));
```

Comment:
I understand why does it look like this, but ternary logic based on boolean looks weird

Reply:
Probably, I need to rewrite this. Looks weird, agree.